zyngine/ctrldev/akai_apc_key25/looper_handler.py

```diff
@@ -0,0 +1,2275 @@
+        index = 3 * (button - BUTTONS.BTN_KNOB_CTRL_VOLUME)
+        if submode == self.MODE_DEFAULT:
+            # 0, 3, 6, 9 => 1 , 4, 7, 10
+            return index + 1
+        if submode == index + 1:
+            # Is 1st submode of this button
+            index = submode + 1
+            if index not in self.MODE_NUMS:
+                return self.MODE_DEFAULT
+            return index
+        if submode == index + 2:
+            # Is 2nd submode of this button
+            return self.MODE_DEFAULT
+        # Special case for default state
+        if (index == (self.MODE_SESSION_SAVE - 1)):
+            return self.MODE_DEFAULT
+        return index + 1
+
+    def handle_mode_buttons(self, button, evtype):
+        if evtype == EV_NOTE_ON:
+            next = self.nextModeNum(button)
+            self.activateSubmode(next)
+        return
+
+    # @todo: only on note on
+    def select_loop_for_button(self, button):
+        row = button - 0x52
+        track = (
+            -1 if row == 0 else shiftedTrack(row, getLoopoffset(self.state))
+        )  # Assuming shifted_track is a method of self
+
+        if track < self.loopcount:
+            self.dispatch(globAction("selected_loop_num", track))
+            self.just_send("/set", ("s", "selected_loop_num"), ("f", track))
+        return
+
+    def handle_rest_of_buttons(self, button, evtype):
+        shifted = getDeviceSetting("shifted", self.state)
+        if hasattr(self._current_handler, 'on_button') and callable(getattr(self._current_handler, 'on_button')) and self._current_handler.on_button(button, evtype, shifted):
+            return
+        
+        if button == BUTTONS.BTN_STOP_ALL_CLIPS:
+            if evtype == EV_NOTE_ON:
+                if getDeviceSetting("shifted", self.state):
+                    self.just_send("/sl/-1/hit", ("s", "mute_off"))
+                else:
+                    self.just_send("/sl/-1/hit", ("s", "mute_on"))
+            else:
+                # self.render_all_pads(state)  # Assuming render_all_pads is a method of self
+                # @todo: simply turn all leds off
+                self.request_feedback(
+                    "/ping", "/pong"
+                )  # Assuming request_feedback is a method of self
+            return
+
+        if button == BUTTONS.BTN_SHIFT:
+            self.dispatch(deviceAction("shifted", evtype == EV_NOTE_ON))
+            return
+
+        if button == BUTTONS.BTN_TRACK_1:
+            if evtype == EV_NOTE_ON and (getDeviceSetting("shifted", self.state)):
+                self.shift_up()  # Assuming shift_up is a method of self
+            return
+
+        if button == BUTTONS.BTN_TRACK_2:
+            if evtype == EV_NOTE_ON and (getDeviceSetting("shifted", self.state)):
+                self.shift_down()
+            return
+
+        return
+
+    def get_sessions(self):
+        mypath = self.SL_SESSION_PATH
+        onlyfiles = [
+            f
+            for f in listdir(mypath)
```

Comment:
This lines throws me an error:

```
FileNotFoundError: [Errno 2] No such file or directory: '/zynthian/zynthian-my-data/presets/sooperlooper/'
``` 

You may need to check if that directory exists before trying to load a session.